ioos/glider-dac
# Use ioos-qc library in  glider_qc.py

Changes introduced to the script:
With all files having an initial list of qartods variables the following functions are no longer needed and have been commented out: 
find_qc_flags
needs_qc
create_qc_variables
apply_primary_qc

The following function were modified or added to adapt the ioos-qc library methodology:
-- Modified:
apply_qc
run_qc
check_needs_qc
-- Added
update_config

The updated glider_qc.py does the following:
 - Read a netcdf file and lists the legacy variables and the associated qartod variables 
 - Use the quality thresholds in the /data/qc_config.yml to calculate the ioos-qc qartod tests results
 - Tested variables with the  following list of qc tests: ['gross_range_test', 
                                                        'spike_test',
                                                        'rate_of_change_test']
- Aggregate the qc test results to use the worst flag (4 = FAIL) if it does exist in any of the tests. The order of the flag selection is: 9= MISSING , 2=NOT-EVAL, 1=PASS, 3=SUSPECT, 4=FAIL.
                                                      

More functionality to verify:
- check the flat line test: it has a "divide by zero" error produced by taking the diff of the time array. 
- check the spike_test it seems it is not working as expected.
- check the xattr of the file to verify if the file has been quality controlled.
